buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/util/Resources.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.hadoop.gradle.util;
```

Comment:
This should really go in `src/main/java`. Looks like the same exists for `WaitForURL.java`. Mixing .groovy and .java in `src/main/groovy` enables Groovy joint-compilation which incurs a compilation time penalty. If there's no need to have `.java` call out to `.groovy` then it's best to separate them. All your Groovy code will have access to your Java code.